sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusSessionAcquirer.java

```diff
@@ -79,44 +114,31 @@ Mono<Session> acquire(String sessionId) {
+
+        return (timeoutRetryDisabled ? acquireSession : acquireSession.timeout(tryTimeout))
             .retryWhen(Retry.from(retrySignals -> retrySignals.flatMap(signal -> {
                 final Throwable failure = signal.failure();
-                logger.atInfo()
-                    .addKeyValue(ENTITY_PATH_KEY, entityPath)
-                    .addKeyValue("attempt", signal.totalRetriesInARow())
-                    .log(sessionId == null
-                        ? "Error occurred while getting unnamed session."
-                        : "Error occurred while getting session " + sessionId, failure);
-
-                if (failure instanceof TimeoutException) {
+                final boolean isTimeoutError = isTimeoutError(failure);
+                if (isTimeoutError && !timeoutRetryDisabled) {
+                    logger.atVerbose()
+                        .addKeyValue(ENTITY_PATH_KEY, entityPath)
+                        .addKeyValue("attempt", signal.totalRetriesInARow())
+                        .log("Broker timeout while acquiring session '{}'.", sessionName(sessionId), failure);
+                    // retry for a session on Schedulers.parallel() and free the QPid thread for other IO.
                     return Mono.delay(Duration.ZERO);
-                } else if (failure instanceof AmqpException
-                    && ((AmqpException) failure).getErrorCondition() == AmqpErrorCondition.TIMEOUT_ERROR) {
-                    // The link closed remotely with 'Detach {errorCondition:com.microsoft:timeout}' frame because
-                    // the broker waited for N seconds (60 sec hard limit today) but there was no free or new session.
-                    //
-                    // Given N seconds elapsed since the last session acquire attempt, request for a session on
-                    // the 'parallel' Scheduler and free the QPid thread for other IO.
-                    //
-                    return Mono.delay(Duration.ZERO);
-                } else {
-                    final long id = System.nanoTime();
-                    logger.atInfo().addKeyValue(TRACKING_ID_KEY, id).log("Unable to acquire a session.", failure);
-                    // The link-endpoint-state publisher will emit error on the QPid Thread, that is a non-blocking Thread,
-                    // publish the error on the (blockable) bounded-elastic thread to free QPid thread and to allow
-                    // any blocking operation that downstream may do.
-                    return Mono.<Long>error(failure)
-                        .publishOn(Schedulers.boundedElastic())
-                        .doOnError(e -> logger.atInfo()
-                            .addKeyValue(TRACKING_ID_KEY, id)
-                            .log("Emitting the error signal received for session acquire attempt.", e));
                 }
+                final long id = System.nanoTime();
```

Comment:
Do we have a test where this is propagated in the sync client?

Reply:
added in the later commit!